reasoner/controller/ConcludableController.java

```diff
@@ -317,6 +348,26 @@ public void setUp() {
                 hubReactive().registerSubscriber(rootSink);
             }
 
+            @Override
+            protected FunctionalIterator<Explanation> rejectMismatchedInference(Explanation explanation) {
+                Set<Variable> conclusionConceptsToCheck = null;
+                if (concludable.isAttribute()) {
+                    conclusionConceptsToCheck = explanation.variableMapping().get(concludable.asAttribute().attribute().id());
+                } else if (concludable.isIsa()) {
+                    conclusionConceptsToCheck = explanation.variableMapping().get(concludable.asIsa().isa().owner().id());
+                }
+                if (conclusionConceptsToCheck != null) {
+                    for (Variable toCheck : conclusionConceptsToCheck) {
+                        if (explanation.conclusionAnswer().concepts().get(toCheck).asThing().isInferred()) {
+                            return Iterators.single(explanation);
+                        } else {
+                            return empty();
+                        }
+                    }
+                }
+                return Iterators.single(explanation);
+            }
+
```

Comment:
This is coded to specifically catch the scenario where: 
1. the rule conclusion is a `has explicit`
2. the attribute concept is actually an inserted fact
3. the concludable only concerns the attribute and not the `has` edge, in which case this answer no longer represents an inferred answer to the concludable

Not rejecting this answer at this stage leads to an extra explanation, which we deem invalid according to our definition of what should be explained (only inferences should be explained).